test/hierarchy_test.cpp

```diff
@@ -1,217 +1,38 @@
-        REQUIRE(points[0].GetGpsTimeVal() < 247570.5);
-        REQUIRE(points[0].R() == 46);
-        REQUIRE(points[0].G() == 60);
-        REQUIRE(points[0].B() == 92);
-        REQUIRE_THROWS(points[0].GetNirVal());
-        REQUIRE_THROWS(points[0].FlagsBitFields());
-        REQUIRE_THROWS(points[0].ReturnsBitFields());
-        REQUIRE_THROWS(points[0].ScannerChannel());
-        REQUIRE_THROWS(points[0].ScanAngle());
-
-        // Setters
-        points[0].X(INT32_MAX);
-        points[0].Y(INT32_MAX);
-        points[0].Z(INT32_MAX);
-        points[0].Intensity(UINT16_MAX);
-        points[0].NumberOfReturns(7);
-        points[0].ReturnNumber(7);
-        points[0].ScanDirFlag(false);
-        points[0].EdgeOfFlightLineFlag(true);
-        points[0].Classification(31);
-        points[0].ScanAngleRank(90);
-        points[0].UserData(UINT8_MAX);
-        points[0].PointSourceId(UINT8_MAX);
-        points[0].SetGpsTimeVal(DBL_MAX);
-        points[0].R(UINT16_MAX);
-        points[0].G(UINT16_MAX);
-        points[0].B(UINT16_MAX);
-        REQUIRE_THROWS(points[0].SetNirVal(UINT16_MAX));
-        REQUIRE_THROWS(points[0].FlagsBitFields(UINT8_MAX));
-        REQUIRE_THROWS(points[0].ReturnsBitFields(UINT8_MAX));
-        REQUIRE_THROWS(points[0].ScannerChannel(3));
-        REQUIRE_THROWS(points[0].ScanAngle(INT16_MAX));
-
-        REQUIRE(points[0].X() == INT32_MAX);
-        REQUIRE(points[0].Y() == INT32_MAX);
-        REQUIRE(points[0].Z() == INT32_MAX);
-        REQUIRE(points[0].Intensity() == UINT16_MAX);
-        REQUIRE(points[0].NumberOfReturns() == 7);
-        REQUIRE(points[0].ReturnNumber() == 7);
-        REQUIRE(points[0].ScanDirFlag() == false);
-        REQUIRE(points[0].EdgeOfFlightLineFlag() == true);
-        REQUIRE(points[0].Classification() == 31);
-        REQUIRE(points[0].ScanAngleRank() == 90);
-        REQUIRE(points[0].UserData() == UINT8_MAX);
-        REQUIRE(points[0].PointSourceId() == UINT8_MAX);
-        REQUIRE(points[0].GetGpsTimeVal() == DBL_MAX);
-        REQUIRE(points[0].R() == UINT16_MAX);
-        REQUIRE(points[0].G() == UINT16_MAX);
-        REQUIRE(points[0].B() == UINT16_MAX);
-    }
-}
-
 TEST_CASE("Packing Test", "[Hierarchy] ")
 {
     GIVEN("A valid input stream")
     {
         fstream in_stream;
         in_stream.open("test/data/autzen-classified.copc.laz", ios::in | ios::binary);
-        auto reader = std::make_shared<copc::io::Reader>(in_stream);
-
-        Hierarchy h(reader);
+        Reader reader(in_stream);
 
         auto key = VoxelKey(0, 0, 0, 0);
-        auto hier_entry = h.FindNode(key);
+        auto hier_entry = reader.FindNode(key);
 
         // Load las::Points
-        auto points = hier_entry->GetPoints();
+        auto points = reader.GetPoints(hier_entry);
 
         // Pack loaded points into stream
         ostringstream oss;
-        hier_entry->PackPoints(points, oss);
+        hier_entry.PackPoints(points, oss);
```

Comment:
```suggestion
        copc::Node::PackPoints(points, oss);
```
Now it's a static function we can call it from the definition